### PR TITLE
Reinstate the cron jobs.

### PR DESCRIPTION
**Pull request by singularo**
Also needs - https://github.com/universityofadelaide/openshift-client/pull/16

### CODE REVIEW

File: web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php [hunk 1]
```diff
@@ -212,15 +212,14 @@ public function createdEnvironment(
...
-          //$this->client->createCronJob(
-          //  $deployment_name . '-' . \Drupal::service('shp_custom.random_string')->generate(5),
-          //  $image_stream['status']['dockerImageRepository'] . ':' . $source_ref,
-          //  $schedule,
-          //  $args_array,
-          //  $volumes,
-          //  $deploy_data
-          //);
+          $this->client->createCronJob(
+            $deployment_name . '-' . \Drupal::service('shp_custom.random_string')->generate(5),
```
Comment (bart88) [inline web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: Inject this.

File: web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php [hunk 2]
```diff
@@ -63,7 +63,7 @@ public function createdProject(string $name, string $builder_image, string $sour
     ];
 
     try {
-      $image_stream = $this->client->generateImageStreamConfig($sanitised_name);
+      $image_stream = $this->client->generateImageStream($sanitised_name);
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: Not sure this is correct. generateImageStreamConfig() is the right method.

see https://github.com/universityofadelaide/openshift-client/blob/develop/src/Client.php#L736

File: web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php (hunk 1 above)
Comment (singularo) [reply web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: another MR - how do i inject to a plugin?

**Comment by singularo** [comment]
Ok, this should be ready now.

File: web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php (hunk 1 above)
Comment (pingers) [reply web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: You'll have to use `implements ContainerInjectionInterface` and implement the create() method.
Something like
```
  public static function create(ContainerInterface $container) {
    return new static($container->get('shp_custom.random_string'));
  }
```

Comment (fenetikm) [reply web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Menu%21LocalActionDefault.php/8.2.x

**Comment by pingers** [comment]
How do we see that cron ran from openshift? Is the only place you can view cronjobs in the monitoring log?

**Comment by pingers** [comment]
:shipit:

**Pull request merged**